tests/testthat/test-get_publication.R

```diff
@@ -17,3 +17,41 @@ test_that("Retrieve data set list for publication", {
     readRDS("testdata/example_publication_datasets.rds")
   )
 })
+
+test_that("Search doesn't return anything it shouldn't", {
+  expect_equal(
+    get_publications(search = "attendance") |>
+      dplyr::filter(
+        !grepl("attendance", title, ignore.case = TRUE),
+        !grepl("attendance", summary, ignore.case = TRUE)
+      ) |>
+      nrow(),
+    0
+  )
+})
+
+
+test_that("Search doesn't return anything it shouldn't", {
+  result <- get_publications(search = "attendance")
+  expect_equal(
+    result |>
+      dplyr::mutate(title_summary = paste(title, summary)) |>
+      dplyr::filter(
+        grepl("attendance", title_summary, ignore.case = TRUE)
+      ) |>
+      nrow(),
+    nrow(result)
+  )
+})
+
+test_that("Search throws an error if the search term is less than 3 characters", {
+  expect_error(
+    get_publications(search = "AP")
+  )
+})
+
+test_that("Search throws an error if the search term is less than 3 characters", {
```

Comment:
```suggestion
test_that("Search throws a warning if the search term is less than 3 characters", {
```

Reply:
Done, just slightly differently.

Reply:
Still think this should say warning not error?